slides/linear_model.md

```diff
@@ -0,0 +1,260 @@
+.pull-right[<img src="../figures/logistic_3D.svg" width="100%">]
+
+???
+Here is an other way of representing our data.
+In this case, X has two dimension x1 and x2.
+The axis correspond to x1, x2
+and the color correspond to the target label y 
+
+---
+# Model complexity
+Salary = *.4 x* education_level + *.2 x* hours-per-week + *.1 x* age + 
+.red[ *.2 x* zodiac_sign + *.1 x* red_socks] + ...  
+
+* A linear model can also overfit. 
+
+The most classic way to limit its complexity is to push coefficients toward small values. Such model is called *Ridge* for regression settings. 
+???
+
+If we have too many parameters w.r.t. number of samples, it's adviced to penalize the parameters of our models.
+
+With weights penalty, we include the value of the model's weights within the objectif function
+So a penalized model should choose lower weights for almost a similar fit
+
+The complexity parameter \alpha controls the amount of shrinkage: the larger the value of \alpha, the greater the amount of shrinkage and thus the coefficients become more robust to collinearity.
+
+---
+# Bias-variance tradeoff: an example
+
+
+.pull-left.shift-left[<img src="../figures/lin_reg_2_points.svg" width="110%">]
+
+---
+# 2 points example
+
+
+.pull-left.shift-left[<img src="../figures/lin_reg_2_points_no_penalty.svg" width="110%">]
+.pull-right[<img src="../figures/lin_reg_2_points_ridge.svg" width="110%">]
+
+.pull-left.shift-left[     Linear regression]
+.pull-right[         Ridge]
+.pull-left.shift-left[     No bias, high variance]
+.pull-right[         low bias, low variance]
+???
+from http://scipy-lectures.org/packages/scikit-learn/index.html#bias-variance-trade-off-illustration-on-a-simple-regression-problem
+
+Left: As we can see, our linear model captures and amplifies the noise in the data. It displays a lot of *variance*.
+
+Right: Ridge estimator regularizes the coefficients by shrinking lightly them to zero
+
+Ridge displays much less variance. However it systematically under-estimates the coefficient. It displays a *biased* behavior.
+
+This is a typical example of bias/variance tradeof: non-regularized estimator are not biased, but they can display a lot of variance. Highly-regularized models have little variance, but high bias. This bias is not necessarily a bad thing: what matters is choosing the tradeoff between bias and variance that leads to the best prediction performance. For a specific dataset there is a sweet spot corresponding to the highest complexity that the data can support, depending on the amount of noise and of observations available.
+
+---
+# 2 points example
+
+
+.pull-left.shift-left[<img src="../figures/lin_reg_2_points_no_penalty_grey.svg" width="110%">]
+.pull-right[<img src="../figures/lin_reg_2_points_ridge_grey.svg" width="110%">]
+
+.pull-left.shift-left[     Linear regression]
+.pull-right[         Ridge]
+.pull-left.shift-left[     No bias, high variance]
+.pull-right[         low bias, low variance]
+???
+
```

Comment:
We should probably add a sentence regarding the test points. I think that the comment of @GaelVaroquaux would just applied as a comment.